File: torc_client/tests/scripts/mapped_function.py
"""Contains a function that will be mapped across workers."""


def run(data: dict):
    """Function to be mapped across workers."""
    assert "val" in data
    return {"params": data, "result": 5, "output_data_path": "/projects/my-project/run1"}


def postprocess(results: list[dict]):
    """Collects the results of workers."""
    total = 0
    paths = []
    for result in results:
        assert "result" in result
        assert "output_data_path" in result
        total += result["result"]
        paths.append(result["output_data_path"])
    return {"total": total, "output_data_paths": paths}

File: julia_client/Torc/test/mapped_function.jl
using Torc

"""
Function to be mapped across workers.
"""
function run_job_function(params::Dict)
    @assert "val" in keys(params)
    return Dict(
        "params" => params,
        "result" => 5,
        "output_data_path" => "/projects/my-project/run1",
    )
end

"""
Collect the results of workers.
"""
function run_postprocess_function(results::Vector{<:Dict})
    total = 0
    paths = String[]
    for result in results
        @assert "result" in keys(result)
        @assert "output_data_path" in keys(result)
        total += result["result"]
        push!(paths, result["output_data_path"])
    end
    return Dict("total" => total, "output_data_paths" => paths)
end

if abspath(PROGRAM_FILE) == @__FILE__
    process_mapped_function_cli_args(run_job_function, run_postprocess_function)
end
